src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/NeuralNetTranches.java

```diff
@@ -0,0 +1,159 @@
+import org.broadinstitute.hellbender.cmdline.programgroups.ShortVariantDiscoveryProgramGroup;
+import org.broadinstitute.hellbender.engine.*;
+import org.broadinstitute.hellbender.exceptions.GATKException;
+import org.broadinstitute.hellbender.exceptions.UserException;
+import org.broadinstitute.hellbender.utils.io.Resource;
+import org.broadinstitute.hellbender.utils.python.PythonScriptExecutor;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Created by sam on 2/5/18.
+ */
+@CommandLineProgramProperties(
+        summary = "Apply tranche filtering",
+        oneLineSummary = "Apply tranche filtering",
+        programGroup = ShortVariantDiscoveryProgramGroup.class
+)
+@DocumentedFeature
+@BetaFeature
+public class NeuralNetTranches extends CommandLineProgram {
+    @Argument(fullName = StandardArgumentDefinitions.INPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.INPUT_SHORT_NAME,
+            doc = "Input VCF file")
+    private String inputVcf = null;
+
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME,
+            shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME,
+            doc = "Output VCF file")
+    private String outputVcf = null;
+
+    @Argument(fullName = "snpTruthVcf", shortName = "stv", doc = "Input file of known common SNP sites.")
+    private String snpTruthVcf = null;
+
+    @Argument(fullName = "indelTruthVcf", shortName = "itv", doc = "Input file of known common INDEL sites.")
+    private String indelTruthVcf = null;
+
+    @Argument(fullName = "scoreKey", shortName = "sk", doc = "Score key.")
+    private String scoreKey = "CNN_1D";
+
+    @Argument(fullName = "samples", shortName = "s", doc = "Maximum number of truth VCF sites to check.")
+    private int samples = 1200;
+
+
+    @Argument(fullName="tranche",
+            shortName="t",
+            doc="The levels of truth sensitivity at which to slice the data. (in percents, that is 1.0 for 1 percent)",
+            optional=true)
+    private List<Double> tranches = new ArrayList<Double>(Arrays.asList(99.9, 99.0, 90.0));
+
+    // Start the Python executor. This does not actually start the Python process, but fails if python can't be located
+    final PythonScriptExecutor pythonExecutor = new PythonScriptExecutor(true);
+    private File tempFile, tempFileIdx;
+    private String trancheString;
+
+    @Override
+    protected void onStartup() {
+        /* check for successful import of libraries */
+        PythonScriptExecutor.checkPythonEnvironmentForPackage("vqsr_cnn");
+        trancheString = " --tranches "+tranches.toString().substring(1, tranches.toString().length()-1).replace(",", "");
+
+        try {
+            final String idxExt = ".tbi";
+            tempFile = File.createTempFile(outputVcf, "_temp.vcf.gz");
+            tempFileIdx = new File(tempFile.getAbsolutePath()+idxExt);
```

Comment:
Is the index for the intermediate file required by the second pass ? Either way, this is leaking the intermediate file since its not deleted. Same for the index.

Reply:
It is required by the tranches script, added .deleteOnExit() for both.